### PR TITLE
dtrace: add page

### PR DESCRIPTION
**Pull request by DEBADRIBASAK**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):**

### CODE REVIEW

File: pages/osx/dtrace.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# dtrace
+
+> Generic front-end to DTrace facility.
+> This command is a simple interface to invoke the D language compiler, retrieve buffered trace data from the DTrace kernel facility, and a set of basic routines to format and print traced data.
+> This requires root privileges.
```
Comment (blueskyson) [inline pages/osx/dtrace.md:5]:
```suggestion
> A simple interface to invoke the D language compiler, retrieve buffered trace, and print traced data from the DTrace kernel facility.
> Generic front-end to DTrace facility, requiring root privileges.
```

File: pages/osx/dtrace.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+> More information: <https://www.unix.com/man-page/osx/1/dtrace/>.
+
+- Set dtrace's target data model which is the target architecture:
```
Comment (blueskyson) [inline pages/osx/dtrace.md:8]:
```suggestion
- Set target data model for a specific architecture:
```

File: pages/osx/dtrace.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+`dtrace -arch {{arch_name}}`
+
+- Claim anonymous tracing state and display the traced data:
```
Comment (blueskyson) [inline pages/osx/dtrace.md:12]:
```suggestion
- Claim [a]nonymous tracing state and display the traced data:
```

File: pages/osx/dtrace.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+`dtrace -a`
+
+- Set principal trace buffer size. Trace buffer size can include any size suffix like k, m, g or t:
+
+`dtrace -b {{buffer_size}}`
```
Comment (blueskyson) [inline pages/osx/dtrace.md:18]:
```suggestion
- Set principal trace buffer size. Supported units are `k`, `m`, `g`, or `t`:

`dtrace -b {{2g}}`
```

File: pages/osx/dtrace.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+- Compile the specified D Program:
```
Comment (blueskyson) [inline pages/osx/dtrace.md:20]:
```suggestion
- Compile the specified D Program [s]ource file:
```

File: pages/osx/dtrace.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+`dtrace -s {{D_script}}`
+
+- Run tghe specified command and exit upon its completion:
+
+`dtrace -c {{path/to/command}}`
```
Comment (blueskyson) [inline pages/osx/dtrace.md:26]:
```suggestion
- Run the specified [c]ommand and exit upon its completion:

`dtrace -c {{command}}`
```

File: pages/osx/dtrace.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+- Specify function name to trace or list (-l option).  The corresponding argument can include any of the probe description  forms like provider:module:function, module:function or function:
```
Comment (blueskyson) [inline pages/osx/dtrace.md:28]:
```suggestion
- Specify [f]unction name to trace or list (-l option). The corresponding argument can include any of the probe description forms like `provider:module:function`, `module:function` or `function`:
```

File: pages/osx/dtrace.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+`dtrace -f {{function}}`
+
+- Grad the process corresponding to specified pid , cache its symbol table and exit upon completion:
```
Comment (blueskyson) [inline pages/osx/dtrace.md:32]:
```suggestion
- Grad the specified  [p]rocess ID, cache its symbol table, and exit upon completion:
```

**Review by blueskyson — COMMENTED** [review]
Thanks for contribution, here are some suggestions.

File: pages/osx/dtrace.md (hunk 4 above)
Comment (kbdharun) [reply pages/osx/dtrace.md:18]: Is `2g` as buffer size a typo or is it a supported unit too?

Comment (blueskyson) [reply pages/osx/dtrace.md:18]: It's just an example of 2 gb.

File: pages/osx/dtrace.md (hunk 1 above)
Comment (DEBADRIBASAK) [reply pages/osx/dtrace.md:5]: Done.

File: pages/osx/dtrace.md (hunk 3 above)
Comment (DEBADRIBASAK) [reply pages/osx/dtrace.md:12]: Done, but why is this necessary?

File: pages/osx/dtrace.md (hunk 2 above)
Comment (DEBADRIBASAK) [reply pages/osx/dtrace.md]: Done.

File: pages/osx/dtrace.md (hunk 4 above)
Comment (DEBADRIBASAK) [reply pages/osx/dtrace.md]: Done

File: pages/osx/dtrace.md (hunk 5 above)
Comment (DEBADRIBASAK) [reply pages/osx/dtrace.md]: Done

File: pages/osx/dtrace.md (hunk 6 above)
Comment (DEBADRIBASAK) [reply pages/osx/dtrace.md]: Done

File: pages/osx/dtrace.md (hunk 7 above)
Comment (DEBADRIBASAK) [reply pages/osx/dtrace.md]: Done

File: pages/osx/dtrace.md (hunk 8 above)
Comment (DEBADRIBASAK) [reply pages/osx/dtrace.md]: Done

File: pages/osx/dtrace.md (hunk 3 above)
Comment (kbdharun) [reply pages/osx/dtrace.md]: Mnemonics are used interchangeably to highlight long and short options in example descriptions.

**Review by kbdharun — APPROVED** [review]
LGTM, Thanks for your contribution.

**Review by gutjuri — APPROVED** [review]

**Pull request merged**